x-pack/plugins/apm/server/routes/latency_distribution/route.ts

```diff
@@ -78,10 +96,24 @@ const latencyOverallTransactionDistributionRoute = createApmServerRoute({
               (fieldValuePair): QueryDslQueryContainer[] =>
                 termQuery(fieldValuePair.fieldName, fieldValuePair.fieldValue)
             ) ?? []),
+            // when using metrics data, ensure we filter by docs with the appropriate duration field
+            ...(searchAggregatedTransactions
+              ? [
+                  {
+                    exists: {
+                      field: getDurationField(
+                        chartType,
+                        searchAggregatedTransactions
+                      ),
+                    },
+                  },
+                ]
+              : []),
           ],
         },
       },
       percentileThreshold,
+      searchAggregatedTransactions,
```

Comment:
do we still need to pass it? one thing that throws me off is that we both make a decision based on `searchAggregatedTransactions` (on line 103) and pass it as a prop to `getOverallLatencyDistribution`. I'd prefer either one or the other, is that an option?

Reply:
The reason it's used in the query as we pass it in, is so that we don't have to deconstruct/reconstruct the query with the `exists` filter appended to it. It seemed less messy than making that decision from within `getOverallLatencyDistribution`. But I can see how it looks convoluted. 

The reason it must be passed to `getOverallLatencyDistribution` regardless is that it's required in downstream calls of `getDurationField` and `getEventType` within the function. Instead, we could pass the event type and duration field to `getOverallLatencyDistribution`, but that would bring us back to the original feedback of those not being the right abstraction.

So to address this i think the best way forward might be to add logic to getOverallLatencyDistribution which appends the `exists` filter to the passed in query.

Reply:
that sounds good to me. as spans will have this same toggle in the future (and I think also have to define it today with this change), can we name it something like `searchAggregatedMetrics`?

Reply:
@ogupte there are still a couple of places where we both pass `searchMetrics` to a function _and_ change some other parameters based on it. Can we move that to one place?